src/bindings/python/flux/kvs.py

```diff
@@ -175,13 +189,16 @@ def __init__(self, flux_handle=None, path=".", handle=None):
         self.pimpl = self.InnerWrapper(flux_handle, path, handle)
 
     def commit(self, flags=0) -> int:
+        """Commit changes to the KVS, must be called after changes are done"""
         return commit(self.fhdl, flags)
 
     def key_at(self, key):
+        """Get full path to KVS key"""
         p_str = self.pimpl.key_at(key)
         return p_str.decode("utf-8")
 
     def exists(self, name):
+        """Evaluate if key exists in the KVS"""
```

Comment:
Should this say "in the KVS directory"?

Reply:
gahhhhhh, I think I am right at the moment, BUT .... this is another "initial path" bug ... it should be "KVS directory"  .... I will push a separate fix.

```
diff --git a/src/bindings/python/flux/kvs.py b/src/bindings/python/flux/kvs.py
index ded222526..38a91814a 100644
--- a/src/bindings/python/flux/kvs.py
+++ b/src/bindings/python/flux/kvs.py
@@ -200,7 +200,7 @@ class KVSDir(WrapperPimpl, abc.MutableMapping):
 
     def exists(self, name):
         """Evaluate if key exists in the KVS"""
-        return exists(self.fhdl, name)
+        return exists(self.fhdl, self._path + name)
 
     def __getitem__(self, key):
         try:
diff --git a/t/python/t0005-kvs.py b/t/python/t0005-kvs.py
index 20efd9a62..344931aeb 100755
--- a/t/python/t0005-kvs.py
+++ b/t/python/t0005-kvs.py
@@ -90,8 +90,15 @@ class TestKVS(unittest.TestCase):
 
     def test_exists_true(self):
         flux.kvs.put(self.f, "thing", 15)
+        flux.kvs.put_mkdir(self.f, "existsubdir")
+        flux.kvs.put(self.f, "existsubdir.thing2", 14)
         flux.kvs.commit(self.f)
         self.assertTrue(flux.kvs.exists(self.f, "thing"))
+        self.assertTrue(flux.kvs.exists(self.f, "existsubdir.thing2"))
+
+        with flux.kvs.get_dir(self.f, "existsubdir") as kd:
+            self.assertTrue(flux.kvs.exists(self.f, "thing2"))
```